disttask/framework/dispatcher/dispatcher.go

```diff
@@ -374,6 +382,30 @@ func GenerateSchedulerNodes(ctx context.Context) ([]*infosync.ServerInfo, error)
 	return serverNodes, nil
 }
 
+func (d *dispatcher) filterByRole(infos []*infosync.ServerInfo) ([]*infosync.ServerInfo, error) {
+	nodes, err := d.taskMgr.GetNodesByRole("background")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nodes) == 0 {
+		nodes, err = d.taskMgr.GetNodesByRole("")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	res := make([]*infosync.ServerInfo, 0, len(nodes))
```

Comment:
how about d.taskMgr.GetNodesByRole("") return 0 nodes?

Reply:
It will fallback to retry dispatch tasks.
How about change the behavior to select all nodes?